ladybug_geometry/_mesh.py

```diff
@@ -0,0 +1,323 @@
+# coding=utf-8
+"""Base class for all Mesh objects."""
+from __future__ import division
+
+
+class MeshBase(object):
+    """Base class for all Mesh objects.
+
+    Properties:
+        vertices
+        faces
+        colors
+        is_color_by_face
+        face_areas
+        area
+        face_centroids
+    """
+
+    @property
+    def vertices(self):
+        """Tuple of all vertices in this geometry."""
+        return self._vertices
+
+    @property
+    def faces(self):
+        """Tuple of all faces in the mesh."""
+        return self._faces
+
+    @property
+    def colors(self):
+        """Get or set a list of colors for the mesh. Will be None if no colors assigned.
+        """
+        return self._colors
+
+    @colors.setter
+    def colors(self, col):
+        if col is not None:
+            assert isinstance(col, (list, tuple)), \
+                'colors should be a list or tuple. Got {}'.format(type(col))
+            if len(col) == len(self.faces):
+                self._is_color_by_face = True
+            elif len(col) == len(self.vertices):
+                self._is_color_by_face = False
+            else:
+                raise ValueError('Number of colors ({}) does not match the number of'
+                                 ' mesh faces ({}) nor the number of vertices ({}).'
+                                 .format(len(col), len(self.faces), len(self.vertices)))
+            if isinstance(col, list):
+                col = tuple(col)
+        self._colors = col
```

Comment:
When colors are None, we keep it as None. There's no good reason to double the memory taken up by the mesh for a case where we are not planning on using colors. I image that this will become particularly helpful for huge butterfly meshes. If it helps better explain this decision, I note the Rhino meshes work the same way. By default, Rhino meshes have no colors assigned to them and you have to actually generate them through an method like CreateMonotoneMesh before you can actually edit the colors.

Reply:
When colors are None, we keep it as None. There's no good reason to double the memory taken up by the mesh for a case where we are not planning on using colors. I image that this will become particularly helpful for huge butterfly meshes. If it helps better explain this decision, I note the Rhino meshes work the same way. By default, Rhino meshes have no colors assigned to them and you have to actually generate them through an method like CreateMonotoneMesh before you can actually edit the colors.